### ISSUE TITLE
[RFC] Add option to strip the env var prefix from config in settings files

### ISSUE THREAD

**Issue by taranlu-houzz** [dynaconf/dynaconf#1170]
**Is your feature request related to a problem? Please describe.**
I prefer to have the env var names exactly match the names used in the config file and application code.

**Describe the solution you'd like**
- Add a param to the dynaconf config that is off by default (e.g. something like `settings_files_strip_prefix`)
- When parsing the settings files, if the config name begins with the prefix set via `envvar_prefix`, strip that (dealing with the following underscore would probably require additional logic or at least clear documentation).
- If the config name doesn't begin with the prefix, just proceed as usual.
- Ultimately, I would like the following ways of setting a config value to be properly merged to the same final attribute on the dynaconf object:
  - env var with prefix: `PREFIX_MY_VAR=123`
  - config name with prefix: in `setting.toml`: `PREFIX_MY_VAR = "123"`
  - config name without prefix: in `settings.toml`: `MY_VAR = "123"`
  - So that the final config value is accesses as `dynaconf_obj.MY_VAR`

**Describe alternatives you've considered**
This isn't an essential feature, just a nice to have.

**Additional context**

**Comment by rochacbruno** [comment]
if you set `envvar_prefix=False` Dynaconf will load all the envvars regardless of prefix, is that what you are looking for?

**Comment by taranlu-houzz** [comment]
@rochacbruno Not quite. Right now, if I enable `envvar_prefix`, it will detect `PREFIX_MY_VAR=123` in the environment and store it as `MY_VAR`, but if I set `PREFIX_MY_VAR = "123"` in the settings file, it will pick that up with that name including the prefix, so I will have `dynaconf_obj.MY_VAR` and dynacof_obj.PREFIX_MY_VAR`.

**Comment by rochacbruno** [comment]
Ok, got it now.

We have prefix [filtering](https://www.dynaconf.com/advanced/#prefix-filtering), but that will just apply the same filtering done to envvars to any source file.

It looks like the feature you are looking for is the ability to set alias #1137

Lets keep this issue open to be considered when implementing the new alias system.
